am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java

```diff
@@ -21,8 +30,42 @@ void setupTest() {
     }
 
     @Test
-    void createQuery_whenCreatingResourceAccess_ResourceAccessAppearsInDatabase() {
-        ams.createResourceAccess(createRecord(resourceId, ACCESSOR_ID, EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS));
+    void emptyPermissionsMapShouldThrowException() {
+        Map<JsonPointer, Set<Permission>> emptyAttributePermissions = new ConcurrentHashMap<>();
+
+        assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(() ->
+            ams.grantExplicitResourceAccess(grantAccess(resourceId, ACCESSOR_ID, emptyAttributePermissions)))
+            .withMessage("Attribute permissions cannot be empty");
+    }
+
+    @Test
+    void whenCreatingResourceAccessResourceAccessAppearsInDatabase() throws TransactionRolledbackException {
+        Map<JsonPointer, Set<Permission>> singleAttributePermission = new ConcurrentHashMap<>();
+        singleAttributePermission.put(JsonPointer.valueOf(""), EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS);
+
+        ams.grantExplicitResourceAccess(grantAccess(resourceId, ACCESSOR_ID, singleAttributePermission));
+
+        assertThat(countResourcesById(resourceId)).isEqualTo(1);
+    }
+
+    @Test
+    void whenCreatingResourceAccessMultipleEntriesAppearInDatabase() throws TransactionRolledbackException {
+        Map<JsonPointer, Set<Permission>> multipleAttributePermissions = new ConcurrentHashMap<>();
+        multipleAttributePermissions.put(JsonPointer.valueOf(""), EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS);
+        multipleAttributePermissions.put(JsonPointer.valueOf("/name"), EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS);
+
+        ams.grantExplicitResourceAccess(grantAccess(resourceId, ACCESSOR_ID, multipleAttributePermissions));
+
+        assertThat(countResourcesById(resourceId)).isEqualTo(2);
+    }
+
+    @Test
+    void whenCreatingDuplicateResourceAccessEntryIsOverwritten() throws TransactionRolledbackException {
+        Map<JsonPointer, Set<Permission>> duplicateAttributePermissions = new ConcurrentHashMap<>();
+        duplicateAttributePermissions.put(JsonPointer.valueOf("/name"), EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS);
```

Comment:
I am not sure this is good test. Are you sure DB is hit twice? Depends how `#hashCode` and `equals` are implemented on `JsonPointer`. It is likely that because maps does not hold duplicates input map has only one entry before it is mapped to record and hit DB.

Callding `ams.grantExplicitResourceAccess` twice with same instance of `ExplicitAccessGrant` will do the work.